torch/utils/_config_module.py

```diff
@@ -326,6 +321,21 @@ def __init__(self, config: _Config):
                     self.env_value_force = env_value
                     break
 
+        # Ensure envvars are boolean
+        if self.justknob is not None:
+            assert isinstance(
+                self.default, bool
+            ), f"justknobs only support booleans, {self.default} is not a boolean"
+            if self.value_type is not None and (
```

Comment:
Indentation typo - I expect you want this one level out :). Probably also why your testcase is failing.

Reply:
FYI - We allow, both env variables and JKs on configs, so you want to check environment variables, whether or not we have a JK config.

Reply:
Fixed in latest version